### PR TITLE
Adds Unicode LF character to line endings for clipboard plaintext

### PR DESCRIPTION
**Pull request by Slapbox**
This addresses https://github.com/ianstormtaylor/slate/issues/1932

I do not believe this will break any existing functionality. In my testing I could not find any issues with this solution so far.

### CODE REVIEW

**Comment by zhujinxuan** [comment]
Does this work with table-cell blocks?

**Comment by Slapbox** [comment]
No it does not, **it also doesn't work with list items. Working on list items now.**

I don't think this method will work with table items, though I'm not sure.

**Comment by zhujinxuan** [comment]
@Slapbox Perhaps we can find the dom by node and check the tagName to solve the table and list problem?  But I am afraid that it is a bit too complex

File: packages/slate-react/src/utils/clone-fragment.js [hunk 1]
```diff
@@ -81,6 +81,17 @@ function cloneFragment(event, value, fragment = value.fragment) {
...
+  
+  //  Builds plaintext object for clipboard line by line using top-level nodes.
+  //  Adds '\u000A' (unicode LF) to end of each line except the last.
+  //  This prevents line breaks from being lost when pasting to targets
+  //  which don't support html input.
+  let plainText = ''
+
+  for (let i = 0; i < contents.childNodes.length; i++) {
+    plainText += contents.childNodes[i].innerText
+    if (i < contents.childNodes.length - 1) plainText += '\u000A'
```
Comment (ianstormtaylor) [inline packages/slate-react/src/utils/clone-fragment.js]: Hey @Slapbox any reason you're using `'\u000A'` instead of just `'\n'`?

Comment (Slapbox) [reply packages/slate-react/src/utils/clone-fragment.js]: No good one. We can change it to `\n`.

I just wanted to be extra-explicit for myself about what characters were in use while I was comparing the unicode outputs of Slate vs other major editors.

**Comment by Slapbox** [comment]
Is there some way to differentiate between these 3 elements for the sake of recursion?

image

I'm trying to expand this to cover cases like content in list items, but I'm not sure how to determine with certainty that an element is a block from the regular DOM. A seemingly easy approach was to add `\n` to the end of each element that wasn't a `span`, but inlines break this idea.

**Comment by Slapbox** [comment]
I'm not really sure.

The serializer requires a `value` object and then peels off the `document` from that.  I looked at how the plain serializer works though and I think it would work, if I can pass it a `Value`.

I'm open to explore this, but I don't want to make a mess of the utility file. I'm not sure what the cleanest method for creating a `Value` is here, one which only encompasses the selection. Suggestions?

https://github.com/ianstormtaylor/slate/blob/master/packages/slate-react/src/utils/clone-fragment.js#L19

**Comment by Slapbox** [comment]
@ianstormtaylor I took your advice and solved the problem very easily that way once I understood how to use `Value.create()`

@zhujinxuan This is what the output for tables looks like in plaintext:

image

File: packages/slate-react/src/utils/clone-fragment.js [hunk 2]
```diff
@@ -82,6 +84,16 @@ function cloneFragment(event, value, fragment = value.fragment) {
 
   attach.setAttribute('data-slate-fragment', encoded)
 
+  //  Creates value from only the selected blocks
+  //  Then gets plaintext for clipboard with proper linebreaks for BLOCK elements
+  //  Via Plain serializer
+  const valFromSelection = Value.create({
```
Comment (isubasti) [inline packages/slate-react/src/utils/clone-fragment.js]: I believe you can do `Value.create({document: fragment})` instead as fragment is a slate's document

Comment (Slapbox) [reply packages/slate-react/src/utils/clone-fragment.js]: This is accurate. I'll make that change now. Thank you

**Comment by zhujinxuan** [comment]
@Slapbox 
What will happen for blocks like
```
<table-cell><paragraph> Text </paragraph></table-cell>
```
and 
```
<list-item><paragraph> Text </paragraph></list-item>
```

`slate-edit-table` and `slate-edit-list` wrap the children in blocks as well.  I am not sure whether it will render an extra `\n` in this case.

**Comment by Slapbox** [comment]
@zhujinxuan I only tested this by manually wrapping table cells with `p` tags, but it seems to make no difference in the output. 

Even if this test is inaccurate (I don't believe it is,) surely the output will be dramatically better than the existing method which has no linebreaks or even spaces between blocks.

**Comment by ianstormtaylor** [comment]
Thanks @Slapbox!

**Comment by Slapbox** [comment]
@ianstormtaylor thanks to you! :heart: Slate

**Pull request merged**